lib/runner/variant-reassignment.js

```diff
@@ -132,47 +171,53 @@ export default class VariantReassignment {
           transactionObject
         )
         await this.transactionService.deleteTransaction(key)
-      } catch (e) {
-        this.logger.error('Could not process unfinished transaction', e)
-        throw e
       }
+    } catch (e) {
+      if (retryCount < constants.MAX_RETRIES && !this._isUnrecoverableError(e))
+        await this._handleUnfinishedTransactionsError(retryCount)
+      else
+        throw e
     }
   }
 
-  async _processProductDraft (productDraft, products) {
-    this.logger.debug(
-      'Processing reassignment for productDraft with name %j',
-      productDraft.name
-    )
-
-    const matchingProducts = await this._selectMatchingProducts(productDraft, products)
+  async _processProductDraft (productDraft, products, retryCount = 0) {
+    try {
+      log(
+        `Processing reassignment for productDraft with name ${JSON.stringify(productDraft.name)}`
+      )
 
-    if (matchingProducts.length === 0)
-      return
+      const matchingProducts = await this._selectMatchingProducts(productDraft, products)
 
-    // select using SLUG, etc..
-    const ctpProductToUpdate = this._selectCtpProductToUpdate(productDraft, matchingProducts)
-    this.logger.debug('Selected ctpProductToUpdate with id "%s"', ctpProductToUpdate.id)
+      if (matchingProducts.length === 0)
```

Comment:
this should be a redundant check right? because we filter out products which do not have any matching products at the beginning.

Reply:
https://github.com/commercetools/commercetools-node-variant-reassignment/pull/48/commits/e9efe3413a1fde384b046dcaab4992de0c32b92f